### PR TITLE
Add index to admin events table to improve performance of admin event…

### LINKED ISSUE #10625: Bad performance of admin event view in large environments
### Describe the bug

The admin event view does load very slowly in large environments with many events in the admin event table.
Loading the view takes > 20 seconds.

See also #895

### Version

16, 17

### Expected behavior

The view is loaded < 2 sec

### Actual behavior

Loading the view takes > 20 s

### How to Reproduce?

Create many entries in admin event table (several 100.000), open the admin view => it's slow

### Anything else?

Adding an index solves, the problem, I will provide a PR to fix this.

### PR DESCRIPTION
**Pull request by ioemat**
… view

Closes #10625

### CODE REVIEW

File: model/jpa/src/main/resources/META-INF/jpa-changelog-18.0.0.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ~ *
+  ~ * Unless required by applicable law or agreed to in writing, software
+  ~ * distributed under the License is distributed on an "AS IS" BASIS,
+  ~ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ * See the License for the specific language governing permissions and
+  ~ * limitations under the License.
+  -->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet author="keycloak" id="18.0.0-KEYCLOAK-10625-IDX_ADMIN_EVENT_TIME">
```
Comment (ahus1) [inline model/jpa/src/main/resources/META-INF/jpa-changelog-18.0.0.xml:20]: The format if the id attribute seems to reference an issue on the old JIRA system as it is prefixed with `KEYCLOAK`. As this is referencing a GitHub issue instead, please use the ID `18.0.0-10625-IDX_ADMIN_EVENT_TIME`

File: model/jpa/src/main/resources/META-INF/jpa-changelog-18.0.0.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        <!-- improve loading time of admin event list -->
+        <createIndex tableName="ADMIN_EVENT_ENTITY" indexName="IDX_ADMIN_EVENT_TIME">
+            <column name="ADMIN_EVENT_TIME" type="DESC"/>
```
Comment (ahus1) [inline model/jpa/src/main/resources/META-INF/jpa-changelog-18.0.0.xml:23]: This index only indexes the column ADMIN_EVENT_TIME, and when I compare it with for example EVENT_ENTITY a similar index indexes also the REALM_ID. 

```
create or replace index IDX_EVENT_TIME
    on keycloak.EVENT_ENTITY (REALM_ID, EVENT_TIME);
``` 

Did you consider adding the REALM_ID as a first column to the index like for the other table?

**Review by ahus1 — CHANGES_REQUESTED** [review]

File: model/jpa/src/main/resources/META-INF/jpa-changelog-18.0.0.xml (hunk 2 above)
Comment (ioemat) [reply model/jpa/src/main/resources/META-INF/jpa-changelog-18.0.0.xml]: You're right, it makes totally sense to add the realm as well, I added it.
I also changed the index to make it as similar as possible to the index in `IDX_EVENT_TIME`

**Review by ahus1 — APPROVED** [review]
Thanks for preparing this PR, looks great!

**Review by ahus1 — APPROVED** [review]

**Comment by ahus1** [comment]
I will run the internal DB regression pipeline to see that this is compatible with all databases currently supported.

**Comment by ahus1** [comment]
Here the results from the test:

| DB | Result |
| -- | -- |
| 'db-mysql mysql80' | :green_circle: |
| 'db-mariadb mariadb103' | :green_circle: |
| 'db-postgres postgresql132' | :green_circle: |
| 'db-postgres postgresql132_schema' | :green_circle: |
| 'db-postgres postgresql101' | :green_circle: |
| 'db-allocator-db-mssql2019 mssql2019' | :yellow_circle: failed due to infrastructure problems (couldn't connect to database) |
| 'db-allocator-db-mariadb mariadb_galera_103' | :green_circle:
| 'db-allocator-db-postgresplus postgresplus131', | :yellow_circle:  failed due to infrastructure problems (couldn't connect to database) |
| 'db-allocator-db-oracleRAC oracle19cRAC' | :yellow_circle: Liquibase succeeded, startup of WildFly distribution failed when creating the realm with "javax.transaction.RollbackException: ARJUNA016102: The transaction is not active!")

@hmlnarik - please let me know if this is sufficient or if I should wait for the mssql2019/postgresplus131 to become available for tests again.

**Comment by hmlnarik** [comment]
@ahus1 postgresplus131 needn't run, but MSSQL should be tested. Could you please try it locally for MSSQL? Check `db-mssql` profile in arquillian testsuite, see [here](https://github.com/keycloak/keycloak/blob/main/docs/tests-db.md#using-built-in-profiles-to-run-database-tests-using-docker-containers) for more details. Running just a single test even on undertow is sufficient - important is that the database creation succeeds.

**Comment by ahus1** [comment]
@hmlnarik - I ran the following and it was successful: 

```
export DOCKER_HOST="unix:/run/user/$(id -u)/podman/podman.sock"
podman system service -t 3600 &
mvn clean install -DskipTests -Pdistribution
mvn clean install -nsu -B -e -f testsuite/integration-arquillian/servers/auth-server -Pauth-server-wildfly
mvn clean install -Dtest=PasswordHashingTest.* -f testsuite/integration-arquillian/tests/base/pom.xml -Pauth-server-wildfly -Pdb-mssql
```

resulting in logs indicating that MS SQL and WildFly where started, and  `BUILD SUCCESS` and `Tests run: 9, Failures: 0, Errors: 0, Skipped: 0`

Therefore I assume this is now ready to be merged. Thanks!

**Review by hmlnarik — APPROVED** [review]
Approved per @ahus1 's approval

**Pull request merged**